rust-src/concordium_base/src/transactions.rs

```diff
@@ -2432,8 +2433,11 @@ pub mod construct {
 
     impl PreAccountTransactionV1 {
         /// Add a sponsor for the transaction.
-        pub fn add_sponsor(&mut self, sponsor: AccountAddress) -> &mut Self {
+        pub fn add_sponsor(&mut self, sponsor: AccountAddress, num_sponsor_sigs: u32) -> &mut Self {
             self.header.sponsor = Some(sponsor);
+            // Add 32 for the sponsor account address and 100 * #(sponsor signatures).
+            self.header.energy_amount =
+                self.header.energy_amount + Energy::from(32 + 100 * u64::from(num_sponsor_sigs));
```

Comment:
```suggestion
                self.header.energy_amount + Energy::from(32 + cost::A * u64::from(num_sponsor_sigs));
```

instead.

Also wondering if we should fail here if a sponsor has already been added? 🤔 Might also make sense to fail if a signature has already been added for the sender, as that'll invalidate that signature?